src/chunk.c

```diff
@@ -836,39 +842,54 @@ ts_chunk_create_base(int32 id, int16 num_constraints)
 	return chunk;
 }
 
+static ScanFilterResult
+chunk_tuple_dropped_filter(TupleInfo *ti, void *arg)
+{
+	ChunkScanResult *chunkresult = arg;
+	bool isnull;
+	Datum dropped = heap_getattr(ti->tuple, Anum_chunk_dropped, ti->desc, &isnull);
+
+	Assert(!isnull);
+	chunkresult->is_dropped = DatumGetBool(dropped);
+
+	return chunkresult->is_dropped ? SCAN_EXCLUDE : SCAN_INCLUDE;
+}
+
 static ScanTupleResult
 chunk_tuple_found(TupleInfo *ti, void *arg)
 {
-	Chunk *chunk = arg;
+	ChunkScanResult *chunkresult = arg;
+
+	if (NULL == chunkresult->chunk)
+		chunkresult->chunk = MemoryContextAllocZero(ti->mctx, sizeof(Chunk));
+
+	chunk_fill(chunkresult->chunk, ti->tuple, ti->desc);
+	chunkresult->is_dropped = chunkresult->chunk->fd.dropped;
 
-	chunk_fill(chunk, ti->tuple, ti->desc);
 	return SCAN_DONE;
 }
 
 /* Fill in a chunk stub. The stub data structure needs the stub ID and constraints set.
  * The rest of the fields will be filled in from the table data. */
 static Chunk *
-chunk_fill_from_stub(Chunk *result, ChunkStub *chunk_stub, bool tuplock)
+chunk_fill_from_stub(ChunkScanResult *result, ChunkStub *chunk_stub)
 {
 	ScanKeyData scankey[1];
 	Catalog *catalog = ts_catalog_get();
 	int num_found;
-	ScannerCtx ctx;
-
-	result->fd.id = chunk_stub->id;
-	result->constraints = chunk_stub->constraints;
-	result->cube = chunk_stub->cube;
-	ctx = (ScannerCtx) {
+	Chunk *chunk;
+	ScannerCtx ctx = {
 		.table = catalog_get_table_id(catalog, CHUNK),
 		.index = catalog_get_index(catalog, CHUNK, CHUNK_ID_INDEX),
 		.nkeys = 1,
 		.scankey = scankey,
 		.data = result,
+		.filter = chunk_tuple_dropped_filter,
```

Comment:
How does this filter work?  Looking at the function, I'd expect that this would result in the scan not returning a dropped chunk.  However, you also added code below to handle the case where a dropped chunk will return NULL instead of hitting an error.

Reply:
Dropped chunks are not returned, but the is_dropped boolean is set in the ChunkStubScanCtx to indicate to the caller that there was metadata marked as dropped.

Not sure this answers your question?